test/src/main/java/org/apache/accumulo/test/GCRunIT.java

```diff
@@ -0,0 +1,242 @@
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Tag(MINI_CLUSTER_ONLY)
+public class GCRunIT extends SharedMiniClusterBase {
+
+  public static final Logger log = LoggerFactory.getLogger(GCRunIT.class);
+
+  @Override
+  protected Duration defaultTimeout() {
+    return Duration.ofSeconds(30);
+  }
+
+  @BeforeAll
+  public static void setup() throws Exception {
+    SharedMiniClusterBase.startMiniCluster();
+  }
+
+  @AfterAll
+  public static void teardown() {
+    SharedMiniClusterBase.stopMiniCluster();
+  }
+
+  /**
+   * Simple run through GCRun candidate to validate that other tests change behaviour when error
+   * injected.
+   */
+  @Test
+  public void goPath() throws Exception {
+    final String[] names = getUniqueNames(2);
+    final String table1 = names[0];
+    final String clone1 = names[1];
+
+    fillMetadataEntries(table1, clone1);
+
+    try (AccumuloClient client = Accumulo.newClient().from(getClientProps()).build()) {
+
+      var context = getCluster().getServerContext();
+
+      scanReferences(new GCRun(Ample.DataLevel.ROOT, context));
+      scanReferences(new GCRun(Ample.DataLevel.METADATA, context));
+      scanReferences(new GCRun(Ample.DataLevel.USER, context));
+
+      client.tableOperations().delete(clone1);
+    }
+  }
+
+  /**
+   * Force deletion of dir entry from the metadata table - expect GCRun to throw exception scanning
+   * candidate
+   */
+  @Test
+  public void forceMissingDirTest() throws Exception {
+
+    final String[] names = getUniqueNames(2);
+    final String table1 = names[0];
+    final String clone1 = names[1];
+
+    fillMetadataEntries(table1, clone1);
+    try (AccumuloClient client = Accumulo.newClient().from(getClientProps()).build()) {
+
+      client.securityOperations().grantTablePermission(getAdminPrincipal(), MetadataTable.NAME,
+          TablePermission.WRITE);
+
+      String cloneId = client.tableOperations().tableIdMap().get(clone1);
+
+      // pick last row that should always exist.
+      final Mutation m = new Mutation(new Text(cloneId + "<"));
+      final Text colf = new Text(MetadataSchema.TabletsSection.ServerColumnFamily.NAME);
+      final Text colq = new Text(DIRECTORY_QUAL);
+      m.putDelete(colf, colq, new ColumnVisibility());
+
+      try (BatchWriter bw = client.createBatchWriter(MetadataTable.NAME,
+          new BatchWriterConfig().setMaxMemory(Math.max(m.estimatedMemoryUsed(), 1024))
+              .setMaxWriteThreads(1).setTimeout(5_000, TimeUnit.MILLISECONDS))) {
+        log.info("forcing delete of srv:dir with mutation {}", m.prettyPrint());
+        bw.addMutation(m);
+      }
+
+      var context = getCluster().getServerContext();
+
+      scanReferences(new GCRun(Ample.DataLevel.ROOT, context));
+      scanReferences(new GCRun(Ample.DataLevel.METADATA, context));
+      // "missing srv:dir prevents full reference scan for user table
+      assertThrows(IllegalStateException.class,
```

Comment:
assertThrows returns the exception, could add additional checks on the returned exceptions message if that would make sense

Reply:
The exception being thrown is the IllegalStateException generated by the new check (line 168 in GCRun).  Or, I don't understand this comment.

Reply:
I think the comment here is that you can do additional asserts on the message itself, rather than just checking the exception type. For example:

```java
  var e = assertThrows(IllegalStateException.class, () -> somethingThatThrows());
  assertEquals(expectedExceptionMessage, e.getMessage());
```